### PR TITLE
kmod: 23 -> 24 and update kmod-blacklist-ubuntu to zesty

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

- kmod was out of date.
- kmod-blacklist-ubuntu was *very* out of date
- kmod-blacklist-ubuntu blacklists i2c_i801 for no good reason

I've tried building them and they all build fine. I haven't done nox-review wip as it causes quite a bit of rebuilding.

@globin, would you consider this for 17.03 or is it too late?

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/os-specific/linux/kmod-blacklist-ubuntu/default.nix [hunk 1]
```diff
@@ -1,13 +1,14 @@
 { stdenv, fetchurl, gnugrep, findutils }:
+
 let
-  version = "3ubuntu1"; # Saucy
-in
-stdenv.mkDerivation {
+  version = "22-1.1ubuntu1"; # Zesty
```
Comment (Mic92) [inline pkgs/os-specific/linux/kmod-blacklist-ubuntu/default.nix]: Given that the blacklist is from 2013 should we also back port it to 17.03?

**Comment by Mic92** [comment]
I would also vote for a backport of the blacklist. The list is supposed to be in sync with the kernel.

**Review by fpletz — APPROVED** [review]

**Comment by vcunat** [comment]
Updating the kmod blacklist broke our X tests (bisected).  Good test case: `nix-build nixos/release.nix -A tests.i3wm.x86_64-linux` &ndash; the X log will end with:
```
(II) modesetting: Driver for Modesetting Kernel Drivers: kms
(EE) open /dev/dri/card0: No such file or directory
(WW) Falling back to old probe method for modesetting
(EE) open /dev/dri/card0: No such file or directory
(EE) Screen 0 deleted because of no matching config section.
(II) UnloadModule: "modesetting"
(EE) Device(s) detected, but none match those in the config file.
(EE)
Fatal server error:
(EE) no screens found(EE)
(EE)
Please consult the The X.Org Foundation support
   at http://wiki.x.org
 for help.
(EE) Please also check the log file at "/dev/null" for additional information.
(EE)
(EE) Server terminated with error (1). Closing log file.
```
(then the service will restart a few times and later the test will wait for a timeout.)

Any ideas why this happens?  Note that we've picked this problem into 17.03 as well /cc @globin.

**Comment by vcunat** [comment]
The correct log opens `/dev/dri/card0` without problems.

**Comment by globin** [comment]
If nobody has a solution until tomorrow evening, I'll revert on the release branch

**Comment by vcunat** [comment]
I decided to revert the commit on master immediately, because no rebuilds are caused and nixos-unstable channel is over two weeks old now.  We can easily re-revert later.

**Comment by globin** [comment]
Re-reverted in 785e45ac06 with a fix

**Comment by vcunat** [comment]
Hmm, I've completely forgotten 9f7d7adb052cee.

**Pull request merged**